### ISSUE TITLE
Add support for other languages

### ISSUE THREAD

**Issue by ajdapretnar** [biolab/orange3-text#306]
##### Text version

3.5.dev-

##### Orange version

0.2.5

##### Expected behavior
Orange supports many key languages.

##### Actual behavior
No support for Latin, (old) Greek, Polish... Poor support for French, German, Spanish, Portuguese. Think about Chinese, Hindu and Arabic as well.

**Comment by gengyabc** [comment]
I wish this addon can support Chinese, and you may use https://github.com/fxsjy/jieba or https://github.com/hankcs/HanLP as the backend

**Comment by ajdapretnar** [comment]
It does support Chinese (UDPipe lemmatization and sentiment analysis). Word embedding is upcoming.

**Comment by ajdapretnar** [comment]
Ah, do you mean for tokenization? We have absolutely no clue how to work with Chinese, so someone else would have to set this up (especially for tests). HanLP doesn't have English documentation, but Jieba seems nice!

**Comment by gengyabc** [comment]
English version of HanLP is here: https://github.com/hankcs/HanLP/tree/master.

**Comment by ajdapretnar** [comment]
Ok, I've checked both.
HanLP is in my opinion a no-go, because it depends on Tensorflow, which is (from what I know) an big dependency. Also, Text already have NLTK, which covers most of HanLP features.

From what I gathered, the only real issue in Chinese text processing is word segmentation. So we would need a specialized Chinese segmenter. This could be added to the new Preprocess Text as a separate option.
We could go with Jieba, but if possible, I'd like to avoid another dependency. Looks like there's a [segmenter](https://textminingonline.com/dive-into-nltk-part-vi-add-stanford-word-segmenter-interface-for-python-nltk) available also in NLTK. Perhaps we could check that first, then fallback to Jieba if NLTK is insufficient.

I would close this issue, as it is too broad and open a separate specific one.

Continued in #536.
